Add Gradle Enterprise to Project

ParaskP7 opened the pull request:
The goal of this PR is to test the `Gradle Enterprise` configuration in order for the GE trial to begin.

To test:

- Verify builds on [gradle.a8c.com](https://gradle.a8c.com/) (ie `Dashboard`, `Build Scan`, etc).
- CI should also publish to GE (ie `Build Scan` urls, etc).
- Additionally, you could also build and test GE locally (ie `Build Scan` urls, etc).

## Regression Notes
1. Potential unintended areas of impact

Build degradation.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` steps above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ParaskP7 committed:
Build: (GE) Set a root project name

Context: The 'rootProject.name' effectively assigns a name to the build
as a whole, which is used in reports like build scans. If the root
project name is not set, the name will be the container directory name,
which can be unstable (i.e. you can check out your project to any
directory).

ParaskP7 committed:
Build: (GE) Update gradle enterprise plugin to the latest 3.9 version

ParaskP7 committed:
Build: (GE) Point gradle enterprise to a8c server

Context: To connect to a Gradle Enterprise instance, you must configure
the location of the Gradle Enterprise server.

ParaskP7 committed:
Build: (GE) Make configuration work when using a self-signed certificate

Context: If your Gradle Enterprise server uses an SSL certificate that
is not trusted by your build’s Java runtime, you will not be able to
publish build scans without explicitly allowing untrusted servers.

Use of this configuration is a security risk as it makes it easier for
a third party to intercept your build scan data. It should only be used
as a short term workaround until the server can be configured with a
trusted certificate.

ParaskP7 committed:
Build: (GE) Configure project to always publish build scans

Context: In order to publish a build scan for every build
(CI and local), you can use the 'publishAlways()' directive.

ParaskP7 committed:
Build: (GE) Enable capturing of task input files for build comparison

Context: Build scans capture hashes of task inputs, to enable
identifying changes to inputs when comparing builds, among other
features. By default, an overall hash value for each task input property
is captured. This enables identifying which properties changed for a
task (e.g. the source or the classpath for Java compilation), but not
which individual files changed. In order to identify file changes, the
paths and content hashes of each individual input file must be captured,
which can be enabled.

ParaskP7 committed:
Build: (GE) Disable build scan background upload for all CI builds

Context: By default, build scans are uploaded in the background after
the build has finished. This allows the build to finish sooner, but can
be problematic in build environments (e.g. ephemeral CI agents) that
terminate as soon as the build is finished, as the upload may be
terminated before it completes. Background uploading should be disabled
for such environments.

ParaskP7 committed:
Build: (GE) Apply the custom user data gradle plugin

This is done in order to capture more data required for the trial.

Context: The Common Custom User Data Gradle plugin for Gradle Enterprise
enhances published build scans by adding a set of tags, links and custom
values that have proven to be useful for many projects building with
Gradle Enterprise.

ParaskP7 committed:
Build: (GE) Remove previous build scan configuration for CI builds

ParaskP7 committed:
Build: (GE) Remove previous build cache configuration for CI builds

jkmassel commented on settings.gradle:
```diff
@@ -1,7 +1,22 @@
 
+gradleEnterprise {
+    server = "https://gradle.a8c.com"
+    allowUntrustedServer = true // Unless a trusted certificate is already configured in GE.
```
We don't need this line, because we have a trusted SSL cert 🙂

ParaskP7 replied:
Great, thank you, I wasn't sure about it so I defaulted to trust the world 😅  will remove it! 🥇

ParaskP7 replied:
Done: 4169fc7a00451e4cdd36cec4610f31d79d88d1d7

jkmassel commented on settings.gradle:
```diff
@@ -114,15 +129,3 @@ if (localBuilds.exists()) {
         }
     }
 }
-
```
If I'm not mistaken, I think we still need to configure the build cache – just pointing at Gradle Enterprise, not at the cache node.

ParaskP7 replied:
Yes, I think you are correct, it is just that atm I am not sure what we should be doing with `caching` in general, like disabling it completely, enable it completely, or replicate the setup we have on `caching` already so that we don't mess up with our current baseline, which is:

- Using the `org.gradle.caching=true` locally, and
- Using the `buildCache { ... }` block, within the `gradleEnterprise { ... }` block this time, for CI only, just pointing to `gradle.a8c.com` this time.

I recommend waiting on confirmation from the Gradle folks on that before proceeding with it. Wdyt?

ParaskP7 committed:
Build: (GE) Disallow untrusted server

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into build/gradle-enterprise

ParaskP7 committed:
Build: (GE) Revert removal of build cash configuration for CI builds

This is done in order to start gatherign GE data on 'trunk', all
other branches, both local and CI environments, based on the latest and
current baseline.

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into build/gradle-enterprise
